### PR TITLE
Optimization and cache

### PR DESCRIPTION
**Pull request by Ophien**
Added caching mechanism and API for the cn5-query to enhance its performance.

### CODE REVIEW

File: api/scheme/conceptnet.scm [hunk 1]
```diff
@@ -146,6 +160,28 @@
...
+
+;;; (cn5-enable-cache)
+;;; This method enables the caching mechanism
+(define (cn5-enable-cache)
+    (set! cn5-use-cache #t)
+)
+
+;;; (cn5-disable-cache)
+;;; This method disables the caching mechanism
+(define (cn5-disable-cache)
```
Comment (andre-senna) [inline api/scheme/conceptnet.scm:175]: Clear cache on disable

File: api/scheme/conceptnet.scm [hunk 2]
```diff
@@ -561,17 +539,65 @@
...
+	    (cog-logger-debug (format "Inserted edges: ~a" inserted_edges))
+	    (cog-logger-debug "END CREATING ATOMESE")
+	
+            ;;; set cache with the answer
+            (set! cn5-cache (acons concept_name answer cn5-cache))
+
+            ;;; returns the created edges list
+            answer            
+        ) ;;; begin
+	;;; (assoc-ref cn5-cache concept_name)
```
Comment (andre-senna) [inline api/scheme/conceptnet.scm]: This is returning an empty list

**Review by andre-senna — CHANGES_REQUESTED** [review]

File: api/scheme/conceptnet.scm (hunk 2 above)
Comment (Ophien) [reply api/scheme/conceptnet.scm]: solve in 9bbd768

File: api/scheme/conceptnet.scm (hunk 1 above)
Comment (Ophien) [reply api/scheme/conceptnet.scm:175]: solved in b5b6dc6

**Review by andre-senna — APPROVED** [review]

**Pull request merged**